lua/base16-colorscheme.lua

```diff
@@ -6,18 +6,43 @@
 
 local M = {}
 
+local function validate_hilight_args(args)
+    if 'string' == type(args) then
+        return true
+    end
+
+    if 'table' ~= type(args) then
+        return 'args must either be a string or a table'
+    end
+
+    return nil ~= (args.guifg or args.guibg or args.gui or args.guisp),
+        'args must contains at least one of: guifg, guibg, gui, guisp'
+end
+
 -- This is a bit of syntactic sugar for creating highlight groups.
 --
 -- local colorscheme = require('colorscheme')
 -- local hi = colorscheme.highlight
 -- hi.Comment = { guifg='#ffffff', guibg='#000000', gui='italic', guisp=nil }
+-- hi.LspDiagnosticsDefaultError = 'DiagnosticError' -- Link to another group
 --
 -- This is equivalent to the following vimscript
 --
 -- hi Comment guifg=#ffffff guibg=#000000 gui=italic
+-- hi! link LspDiagnosticsDefaultError DiagnosticError
 M.highlight = setmetatable({}, {
     __newindex = function(_, hlgroup, args)
-        local guifg, guibg, gui, guisp = args.guifg, args.guibg, args.gui, args.guisp
+        vim.validate {
+            hlgroup = { hlgroup, 'string' },
+            args = { args, validate_hilight_args, 'string or table' },
+        }
```

Comment:
Since this isn't meant to be called by anyone other than this plugin, I would prefer to not include validation.

Reply:
In this case maybe I should also remove it from the `M` table so it will not be exposed ?

Reply:
Yea that works.

Reply:
I removed it from the module in a separate commit after all, because I saw a comment (removed in the last commit) with an example of how to use the helper by requiring the module.
So it will be easy to revert if you change your mind.